src/Collection.php

```diff
@@ -415,15 +415,14 @@ public static function fromIterable(iterable $iterable): self
 
     /**
      * @param resource $resource
+     *
+     * @return self<int, string>
```

Comment:
@drupol I see a mix of annotations in this project, but I'm not sure why that is. Should we use the psalm ones or the "normal" ones everywhere?

A. `@param`, `@return`, `@template`
B.  `@psalm-param`, `@psalm-return`, `@psalm-template`

For the analysers it doesn't matter because they support both. And in an IDE like PHPStorm / IntelliJ both are properly highlighted.

I personally favour **A** just because of a couple small reasons:
- less characters to write. And easier to mentally parse when reading 😄 
- doesn't tie the code to a specific developer tool (though, again, doesn't really matter because both are supported)

Reply:
This is also on my Todo list! Option A for the win :)